File: data-structures/java/Stack.java
public class Stack<T>
{
    private int size = 0;
    private T defaultValue;
    private SingleLinkNode<T> top = null;

    public Stack() {}

    public void push(T data)
    {
        this.size++;
        SingleLinkNode<T> node = new SingleLinkNode<T>(data, this.top);
        this.top = node;
    }
    
    public void pop()
    {
        if(this.top == null) return;

        this.size--;
        SingleLinkNode<T> node = this.top;
        this.top = this.top.getNext();
        node.setNext(null);
    }
    
    public T getTop()
    {
        if(this.top != null) return this.top.getData();

        return this.defaultValue;
    }
    
    public boolean isEmpty()
    {
        return this.top == null;
    }

    public int getSize()
    {
        return this.size;
    }
}

File: data-structures/c#/Stack.cs
namespace DataStructures
{
	public class Stack<T>
	{
		private class Node
		{
			public T data;
			public Node next;

			public Node(T data, Node next = null)
			{
				this.data = data;
				this.next = next;
			}
		}

		private long count = 0;
		private Node top = null;

		public Stack()
		{}

		~Stack()
		{
			Delete();
		}

		public void Push(T data)
		{
			count++;
			Node node = new Node(data, top);
			top = node;
		}

		public void Pop()
		{
			if(top == null) return;

			count--;
			Node node = top;
			top = top.next;
			node.next = null;
		}

		public T Top()
		{
			if(top == null) return default(T);

			return top.data;
		}

		public bool Empty()
		{
			return top == null;
		}

		public long Count()
		{
			return count;
		}

		public void Delete()
		{
			while(top != null)
			{
				count--;
				top = top.next;
			}
		}
	}
}